### PR TITLE
NOISSUE - Update docs to include ws-adapter

### PR DESCRIPTION
**Pull request by AryanGodara**
Signed-off-by: aryan <aryangodara03@gmail.com>

### Which issue(s) does this PR fix/relate to?
NOISSUE

### List any changes that modify/break current functionality
N/A

### Have you included tests for your changes?
NO

### Did you document any new/modified functionality?
YES

### Notes
N/A

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Mainflux IoT platform is comprised of the following services:
 | [things](https://github.com/mainflux/mainflux/tree/master/things)         | Manages platform's things, channels and access policies                          |
 | [http-adapter](https://github.com/mainflux/mainflux/tree/master/http)     | Provides an HTTP interface for accessing communication channels                  |
 | [mqtt-adapter](https://github.com/mainflux/mainflux/tree/master/mqtt)     | Provides an MQTT and MQTT over WS interface for accessing communication channels |
+| [ws-adapter](https://github.com/mainflux/mainflux/tree/master/ws)         | Provides a WebSocket interface for accessing communication channels     |
```
Comment (dborovcanin) [inline docs/architecture.md:13]: Fix `|` alignment.

File: docs/messaging.md [hunk 2]
```diff
@@ -66,8 +66,42 @@ The most of the notifications received from the Adapter are non-confirmable. By
...
+`/ws/channels/<channel_id>/messages?authorization=<thing_auth_key>&content-type=<content-type>`.
+
+### Basic nodejs example
+
+```javascript
+const WebSocket = require('ws');
+// do not verify self-signed certificates if you are using one
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
+// cbf02d60-72f2-4180-9f82-2c957db929d1  is an example of a thing_auth_key
+const ws = new WebSocket('wss://localhost/ws/channels/1/messages?authorization=cbf02d60-72f2-4180-9f82-2c957db929d1&content-type=application%2Fsenml%2Bjson')
```
Comment (manuio) [inline docs/messaging.md:90]: Can we use `wss://localhost/ws/channels/<channel_id>/messages?authorization=<thing_key>&content-type=application/senml+json` ?

**Review by manuio — CHANGES_REQUESTED** [review]

File: docs/authentication.md [hunk 3]
```diff
@@ -80,6 +80,32 @@ Then, you can create user and provision things and channels. Now, in order to se
...
+})
+ws.on('message', (data) => {
+    console.log(data)
+})
+ws.on('error', (e) => {
+    console.log(e)
+})
+```
+
+As you can see, `Authorization` header does not have to be present in the HTTP request, since the key is present in the certificate. However, if you pass `Authorization` header, it _must be the same as the key in the cert_. In the case of MQTTS, `password` filed in CONNECT message _must match the key from the certificate_. In the case of WSS, `Authorization` header or `authorization` query parameter _must match cert key_.
```
Comment (dborovcanin) [inline docs/authentication.md:107]: Moze this section up, under **Mutual TLS Authentication with X.509 Certificates**.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: docs/messaging.md (hunk 2 above)
Comment (AryanGodara) [reply docs/messaging.md:90]: No, it returns an error with this url. Connection failed.
`1006 Abnormal Closure: No close frame was received.`

**Comment by manuio** [comment]
@AryanGodara Why you closed this one?

**Pull request closed without merging**